src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -352,7 +353,6 @@ public Builder addJavadoc(String format, Object... args) {
     }
 
     public Builder addAnnotations(Iterable<AnnotationSpec> annotationSpecs) {
-      checkState(anonymousTypeArguments == null, "forbidden on anonymous types.");
```

Comment:
Happy to be a bit cleverer here if you have some ideas how, but AFAICT I don't know at this point how I'll be used, just that I'm an anonymous class.  That means I can't limit annotation usage to enums only.